ceph/ceph_admin/helper.py

```diff
@@ -2,8 +2,216 @@
+        "node-exporter",
+        "prometheus",
+    ]
+
+    def __init__(self, node, cluster, specs):
+        """
+        Initialize the PrepareSpec
+
+        Args:
+            node: ceph node where spec file to be created
+            cluster: ceph cluster (ceph-nodes)
+            specs: service specifications
+
+        specs:
+          - service_type: host
+            address: true
+            labels: apply-all-labels
+            nodes:
+                - node2
+                - node3
+          - service_type: mon
+            placement:
+              nodes:
+                - node2
+                - node3
+          - service_type: mgr
+            placement:
+                count: 2
+          - service_type: alertmanager
+            placement:
+                count: 1
+          - service_type: crash
+            placement:
+                host_pattern: '*'
+          - service_type: grafana
+            placement:
+                count: 1
+          - service_type: node-exporter
+            placement:
+                host_pattern: '*'
+          - service_type: prometheus
+            placement:
+                count: 1
+        """
+        self.cluster = cluster
+        self.node = node
+        self.specs = specs
+        self.template_path = dirname(__file__) + "/jinja_templates/"
+
+    @staticmethod
+    def get_hostname(node):
+        return node.shortname
+
+    @staticmethod
+    def get_addr(node):
+        return node.ip_address
+
+    @staticmethod
+    def get_labels(node):
+        return node.role.role_list
+
+    def _get_template(self, service_type):
+        """
+        Return Jinja template based on the service_type
+        Args:
+            service_type: service name (ex., "host")
+        Returns:
+            template
+        """
+        path = self.template_path + f"{service_type}.jinja"
+        with open(path) as fd:
+            template = fd.read()
+        return Template(template)
+
+    def render_hosts(self, spec):
+        """
+        Return hosts spec content based on host config
+        args:
+            spec: hosts specification
+
+        spec:
+          - service_type: host
+            address: true
+            labels: apply-all-labels
+            nodes:
+                - node2
+                - node3
+        Returns:
+            hosts_spec
+        """
+        template = self._get_template("host")
+        hosts = []
+        address = spec.get("address")
+        labels = spec.get("labels")
+        for node_name in spec["nodes"]:
+            host = dict()
+            node = get_node_by_id(self.cluster, node_name)
+            host["hostname"] = self.get_hostname(node)
+            if address:
+                host["address"] = self.get_addr(node)
+            if labels:
+                host["labels"] = self.get_labels(node)
+            hosts.append(host)
+
+        return template.render(hosts=hosts)
+
+    def render_common_services(self, spec):
```

Comment:
Would it make sense to rename this method to `generate_generic_spec`?

Reply:
Done.